pilot/api/data.py

```diff
@@ -648,8 +648,8 @@ def resolve_protocols(self, files, activity):
 
             protocols = []
             for aname in activity:
-                aname = get_ddm_activity(aname)
-                protocols = ddm.arprotocols.get(aname)
+                aname = get_ddm_activity(aname, self.infosys.queuedata, fdat.ddmendpoint, ddm)
```

Comment:
here is the equivalent of your logic: 

adding one extra line loop  over preferred list of ddm activities is enough (however I did not get why you need to check astorages???)
LAN protocols come first so that Pilot prefers to use LAN protocols if they exists.
```
activity_map = {'es_events': ['write_lan', 'write_wan'],
                'es_failover': ['write_lan', 'write_wan'],
                'es_events_read': ['read_lan', 'read_wan']}

for aname in activity:
    for aname in amap.get(aname) or [aname]:
        protocols = ddm.arprotocols.get(aname)
        if protocols:  ## oor add and self.infosys.queuedata and fdat.ddmendpoint in self.infosys.queuedata.astorages.get(aname, {}):
            break
```